packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -39,11 +39,12 @@ export function exceptionToResponse(err: any): IResponse {
     // Capture error objects, not stack itself, as stack retrieval is very expensive operation, so we delay it
     const errWithStack = generateErrorWithStack();
 
+    const { message, stack } = extractLogSafeErrorProperties(err, false);
     return {
         mimeType: "text/plain",
         status,
-        value: `${err}`,
-        get stack() { return ((err?.stack) as (string | undefined)) ?? errWithStack.stack; },
+        value: message,
+        get stack() { return stack ?? errWithStack.stack; },
```

Comment:
Consider calling `generateErrorWithStack()` here directly so its not called unnecessarily if there is already a stack.

Reply:
Actually that will change the callstack, it's correct as-is.